### PR TITLE
[TRAFODION-3306] Add the TRUNC Function in the Trafodion SQL Reference Manual

### PR DESCRIPTION
**Pull request by liuyu000**
@DaveBirdsall Could you please help review? Many thanks.

### CODE REVIEW

File: docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc [hunk 1]
```diff
@@ -12460,6 +12460,122 @@ CHAR(20). This expression uses the TRIM function to return the value
 TRIM(first_name) || ' ' || TRIM (last_name)
 ```
 
+[[trunc_function]]
+== TRUNC Function
+
+The TRUNC (TRUNCATE) function truncates `_numeric-expression_` on an optional `_decimal-place_` decimal position(s), that is, it rounds down `_numeric-expression_` to the nearest (or equal) integer closer to zero *without rounding*.
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: The second argument makes this description awkward. If the second argument were zero, then we are rounding down to the nearest integer closer to zero. If the second argument is positive. we aren't necessarily rounding to an integer. TRUNC(1.23,1) gives 1.2, and 1.2 isn't an integer. And I'm not sure what happens when the second argument is negative. Maybe TRUNC(123,-1) gives 120?

If this is true, one way to describe this is TRUNC is like ROUND, except it always rounds toward zero.

Here is a possible wordsmith (please do a bit of experimentation to see if this is right);

The TRUNC (TRUNCATE) function truncates '_numeric-expression_'. If the second argument is not supplied, or is supplied and is zero, this rounds the '_numeric-expression_' to the nearest integer closer to zero. If the second argument is supplied and is positive, say, n, it rounds the '_numeric_expression_' to the nearest multiple of 10^-n that is closer to zero. In these two cases, decimal places to the right of the decimal point are effectively deleted. If the second argument is supplied and is negative, say, -n, it rounds '_numeric_expression_' to the nearest multiple of 10^n. Any decimal places to the right of the decimal point are effectively deleted.

File: docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc [hunk 2]
```diff
@@ -12460,6 +12460,122 @@ CHAR(20). This expression uses the TRIM function to return the value
...
+For more information, see <<numeric_value_expressions,Numeric Value Expressions>>.
+
+* `_decimal-place_`
+
++
+is an integer specifying the precision of the truncation.
+
+NOTE: `_numeric-expression_` and `_decimal-place_` cannot be NULL.
+
+NOTE: Currently, `_decimal-place_` can be zero or a positive integer, while *negative integer* is *not supported*.
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: Ah, this makes it easier. We don't have to worry about the negative case for the second argument.

File: docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc [hunk 3]
```diff
@@ -12460,6 +12460,122 @@ CHAR(20). This expression uses the TRIM function to return the value
...
+  &#9702; `_decimal-place_` is not specified 
+
+| *TRUNC* function = *CEIL* function
+
+|===
+
+* The TRUNC function is similar to the ROUND function in terms of *reducing the number of decimal places*. 
+
++
+However, the TRUNC function *does not perform any rounding* as the ROUND function does.
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: I would say this differently. I would say that TRUNC rounds toward zero. While ROUND rounds to the nearest integer.

Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: @DaveBirdsall Thank you, I've incorporated your comment.

File: docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc (hunk 2 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: For most databases, TRUNC(123,-1) gives 120, while Trafodion doesn't support negative _decimal-place_.

File: docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc (hunk 1 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc]: For most databases, TRUNC(123,-1) gives 120, while Trafodion doesn't support negative decimal-place.
Your suggestions are correct, I've incorporated them. Thanks again.

**Pull request merged**